File: 2025-redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.cpp
class Solution {
public:
    bool makeEqual(vector<string>& a) {
        vector<int>map(27);
        for(auto x:a)
        for(auto y:x)
        map[y-'a']++;

        int n=a.size();

        for(auto x:map)
        if(x%n!=0) return 0;

        return 1;
        
    }
};

File: 2025-redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.java
class Solution {
    public boolean makeEqual(String[] a) {
        int[] map = new int[27];

        for (String x : a)
            for (char y : x.toCharArray())
                map[y - 'a']++;

        int n = a.length;

        for (int x : map)
            if (x % n != 0) return false;

        return true;
    }
}